x-pack/docs/en/rest-api/security/update-api-key.asciidoc

```diff
@@ -1,5 +1,270 @@
 [role="xpack"]
 [[security-api-update-api-key]]
-=== Update API key information API
+=== Update API key API
 
-coming::[8.4.0]
+++++
+<titleabbrev>Update API key</titleabbrev>
+++++
+
+[[security-api-update-api-key-request]]
+==== {api-request-title}
+
+`PUT /_security/api_key/<id>`
+
+[[security-api-update-api-key-prereqs]]
+==== {api-prereq-title}
+
+* To use this API, you must have at least the `manage_own_api_key` cluster privilege.
+Users can only update their own API keys, i.e., API keys they created.
+To update another user's API key, you must use the run as feature.
+For more information, see <<run-as-privilege>>.
+
+IMPORTANT: It's not possible to use an API key as the authentication credential for this API.
+To update an API key, user password or <<security-api-get-token, token credentials>> are required.
+
+[[security-api-update-api-key-desc]]
+==== {api-description-title}
+
+The API keys created by <<security-api-create-api-key,create API Key>> or <<security-api-grant-api-key,grant API Key>> can be updated using this API.
+It's not possible to update expired API keys, or API keys that have been invalidated by <<security-api-invalidate-api-key,invalidate API Key>>.
+
+This API supports updates to an API key's access scope, i.e., its permissions, and its metadata.
+Other properties such as `name` or `expiration` cannot be updated.
+
+IMPORTANT: An API key's permissions are always restricted by the permissions of the authenticated user creating or updating the API key.
+Any call to <<security-api-update-api-key,update API Key>>, **regardless** of whether parameters are supplied or not, will automatically apply the authenticated user's current permissions to the API key.
+This may result in changes in the API key's access scope, even if you don't specify new <<security-api-update-api-key-api-key-role-descriptors,`role_descriptors`>> in the update request.
+
+[[security-api-update-api-key-path-params]]
+==== {api-path-parms-title}
+
+`id`::
+(Required, string) The ID of the API key to be updated.
+
+[[security-api-update-api-key-request-body]]
+==== {api-request-body-title}
+
+The request body is optional.
+Following parameters can be specified in it:
+
+[[security-api-update-api-key-api-key-role-descriptors]]
+`role_descriptors`::
+(Optional, array-of-role-descriptor) An array of new role descriptors for this API key.
+This parameter is optional.
+When the parameter is supplied but empty, the API key's new permissions will be the current _point in time snapshot of permissions of the authenticated user_.
```

Comment:
If we separate the two following two discussions, it might help make things easier to explain:
1. The API key's effective permission is *always* the intersection of its own privileges and owner user's snapshotted privileges
2. How the privileges can be updated, i.e. a) Always (auto) updating owner user's privileges b) optionally update the key's own privileges. 

Item 2 is what we really care here. But we have to explain 1 because currently we have no overview doc for API keys. We can keep 1 simple to just state the rule is always intersection (no matter how its updated or captured). 

For 2, the tricky bit is the `empty` `role_descriptors` means inherting owner user's privileges. Otherwise, it is not that complicated. So I think we can state the simple rule (optional update) first and leave the special `empty` input as a callout. 

This should let us avoid mentioned things like "previously assigned ..." and not trying to list all combinations.

Reply:
Absolutely up for simplifying. I provided the "verbose" full context here so that we have the complete picture for discussion and it's easier to talk about what should and shouldn't be included.

> The API key's permissions are a combination of it's privileges, plus the owner user's current permissions at the current time when the request is issued.

Is still a little confusing since we talk about the API key's permissions and the API key's privileges which are only very subtly different (if at all). I think using "assigned permissions" or "assigned privileges" works better here.

> The API key's effective permission is always the intersection of its own privileges and owner user's snapshotted privileges

Is not exactly right since if the assigned privileges are empty, it's not the intersection anymore (it would be empty) so we still need some sort of qualifier here.

What about s.th. like:

> The API key's permissions are an intersection of its assigned privileges, plus the owner user's current permissions at the time the request is issued. If no privileges are assigned, the API key has the owner user's full current permissions.

Reply:
> However, I think that we need an example to clarify and solidify this statement.

What if we link to the example requests further down? I've intentionally included examples that highlight the nuances of this.